lua/wire/wireshared.lua

```diff
@@ -168,6 +168,17 @@ local table = table
 local pairs_sortvalues = pairs_sortvalues
 local ipairs_map = ipairs_map
 
+-- WireLib.UnpackWithNils(t) does the same as unpack(t), but returns as many
+-- arguments as table.maxn(t) (instead of as many arguments as #t). This means
+-- it handles unpacking arrays with nil in them as expected.
+function WireLib.UnpackWithNils(t)
+	local function step(i, n)
+		if i > n then return end
+		return t[i], step(i + 1, n)
+	end
+	return step(1, table.maxn(t))
```

Comment:
`table.maxn` will only return the highest index that contains a non nil value, so if the last few indexes of `t` are nil then they aren't included. I think a better solution would be to include an `n` field in the `GetActionInputs` return so you can pass it to `unpack` (`unpack(t, 1, t.n)`).

Reply:
`table.maxn will only return the highest index that contains a non nil value, so if the last few indexes of t are nil then they aren't included.`

a value being nil is the same as it not existing at all so what does this sentence even mean

Reply:
This is being used to unpack `GetActionInputs` which can include nil values, I'd assume it could include them at the end as well.

Reply:
Yes but, including nil values or not at the end of a table or at the end of a function argument list makes no difference whatsoever.

Reply:
Fair enough.

Reply:
Ah, I forgot that `unpack` took an optional index range! That's a much neater solution. (And `WireLib.UnpackWithNils` is then just `unpack(t, 1, table.maxn(t))`, so not worth keeping around as a function that might be useful some time.)